tsdb/meta.go

```diff
@@ -989,6 +993,12 @@ func (m *Measurement) idsForExpr(n *influxql.BinaryExpr) (SeriesIDs, influxql.Ex
 		// that are missing the tag.
 		empty := re.Val.MatchString("")
 
+		var sz int
+		for _, v := range tagVals {
```

Comment:
I don't think this is the correct place to put this. A bunch of the sections right under this one would not use this pre-allocated array. For example, the ones that use `newEvictSeriesIDs` would not use this and it would be a wasted allocation. It looks like this mostly helps with two scenarios. When you use `NEQREGEX` and it includes the empty string and `EQREGEX` and it does not include the empty string.